File: Bit Manipulation/9. Compute 2's Complement.py
# Compute 2's Complement

def twos_complement(num, num_size):

    ones_complement = num ^ ((1 << num_size) - 1)
    twos_comp = ones_complement + 1

    return twos_comp

num = 0b1101
num_size = 4
two_c = twos_complement(num, num_size)
print(bin(two_c).replace("0b", ""))

File: Bit Manipulation/C/9. Compute 2's complement.c
#include<stdio.h>

int twosComp(int n)
{
    int ones_comp = ~n;
    int twos_comp = ones_comp + 1;
    return twos_comp;
}

int main()
{
    int n = 4;
    printf("%d", twosComp(n));
    return 0;
}
